plugins/needs_restarting.py

```diff
@@ -40,6 +40,113 @@
 NEED_REBOOT = ['kernel', 'glibc', 'linux-firmware', 'systemd', 'dbus',
                'dbus-broker', 'dbus-daemon']
 
+class DirectoryOption(object):
+    """ This class handles a directory containing file options
+        This class can sanitize dir path
+    """
+    def __init__(self, path=None):
+        self._path = self._parse(path)
```

Comment:
Since you're only calling `_parse` from here and it's also the only thing in `__init__()` the indirection seems unnecessary, you could put the parsing code here directly. Though see the whole review first, the whole class doesn't seem to be really needed.